### ISSUE TITLE
Update S2365: Rule should report on new collection

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-dotnet#2222]
### Description

The S2365 rule should be improved to start raising on cases where the property returns a new collection from one of the field collection.

```csharp
public List<string> People => new List<string>(_people);
```

as we currently report on

```csharp
public List<string> People => _people.ToList();
```

**Comment by andrei-epure-sonarsource** [comment]
https://rules.sonarsource.com/csharp/RSPEC-2365

**Comment by sm-g** [comment]
@Evangelink 
Please add same rule for ToDictionary/ToArray/ToLookup/ToHashSet etc

**Comment by mary-georgiou-sonarsource** [comment]
This issue will be tracked in [NET-1261](https://sonarsource.atlassian.net/browse/NET-1261) (not publicly accessible)
